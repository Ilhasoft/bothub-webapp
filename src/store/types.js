export default {
  SET_TOKEN: 'SET_TOKEN',
  OPEN_LOGIN_MODAL: 'OPEN_LOGIN_MODAL',
  CLOSE_LOGIN_MODAL: 'CLOSE_LOGIN_MODAL',
  SET_PROFILE: 'SET_PROFILE',
  SET_NICKNAME_AUTHENTICATED: 'SET_NICKNAME_AUTHENTICATED',
  SET_REPOSITORY_RELATED_UUID: 'SET_REPOSITORY_RELATED_UUID',
  CACHE_FETCH: 'CACHE_FETCH',
};
