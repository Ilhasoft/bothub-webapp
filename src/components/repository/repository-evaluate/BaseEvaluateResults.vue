<template>
  <div
    v-if="resultId">
    <div>
      <evaluate-result-example-list
        :id="resultId"
        :query="query" />
      <graphics-result :chart-data="resultsData"/>
    </div>
  </div>
  <p v-else>{{ $t('webapp.evaluate.not_have_results') }}</p>
</template>

<script>
import { mapActions } from 'vuex';
import GraphicsResult from '@/components/repository/repository-evaluate/results/GraphicsResult';
import EvaluateResultExampleList from '@/components/repository/repository-evaluate/results/EvaluateResultExampleList';
import { exampleSearchToDicty, exampleSearchToString } from '@/utils/index';


export default {
  name: 'BaseEvaluateResults',
  components: {
    GraphicsResult,
    EvaluateResultExampleList,
  },
  props: {
    repository: {
      type: Object,
      required: true,
    },
    resultId: {
      type: Number,
      default: null,
    },
    filterByLanguage: {
      type: String,
      default: '',
    },
  },
  data() {
    return {
      querySchema: {},
      query: {},
      resultsData: {},
    };
  },
  watch: {
    filterByLanguage() {
      this.onSearch();
    },
  },
  mounted() {
    this.getResults();
  },
  methods: {
    ...mapActions([
      'getResultsData',
    ]),
    async getResults() {
      if (this.resultId) {
        const append = await this.getResultsData({
          repositoryUuid: this.repository.uuid,
          resultId: this.resultId,
        });
        this.resultsData = append.data;
      }
    },
    onSearch(value) {
      Object.assign(this.querySchema, value);

      if (!this.querySchema.intent) {
        delete this.querySchema.intent;
      }
      if (!this.querySchema.entitie) {
        delete this.querySchema.entitie;
      }
      if (!this.querySchema.label) {
        delete this.querySchema.label;
      }
      const formatedQueryString = exampleSearchToString(this.querySchema);
      this.query = exampleSearchToDicty(formatedQueryString);
      this.query.language = this.filterByLanguage;
    },
  },
};
</script>
