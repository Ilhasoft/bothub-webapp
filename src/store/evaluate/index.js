import actions from './actions';
import state from './state';
import mutations from './mutations';
import getters from './getters';
import types from './types';

export default {
  actions,
  state,
  mutations,
  getters,
  types,
};
