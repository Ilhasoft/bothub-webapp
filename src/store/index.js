import Vue from 'vue';
import Vuex from 'vuex';

import Auth from './auth';
import User from './user';
import Org from './org';
import Repository from './repository';
import Category from './category';
import Example from './example';
import EvaluateExample from './evaluate-example';
import Translate from './translate';
import CachedFetch from './cached-fetch';
import Entity from './entity';
import Intent from './intent';
import News from './news';
import Tutorial from './tutorial';
import External from './external';

Vue.use(Vuex);

const store = new Vuex.Store({
  modules: {
    Auth,
    User,
    Org,
    Repository,
    Category,
    Example,
    Translate,
    CachedFetch,
    EvaluateExample,
    Entity,
    News,
    Tutorial,
    External,
    Intent,
  },
});

store.dispatch('retriveAuthToken');

export default store;
