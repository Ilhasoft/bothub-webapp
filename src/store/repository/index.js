import state from './state';
import getters from './getters';
import actions from './actions';
import mutations from './mutations';
import types from './types';

export default {
  state,
  getters,
  actions,
  mutations,
  types,
};
