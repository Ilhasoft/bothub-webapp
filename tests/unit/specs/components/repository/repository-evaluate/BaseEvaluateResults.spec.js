import { shallowMount, createLocalVue } from '@vue/test-utils';
import BaseEvaluateResults from '@/components/repository/repository-evaluate/BaseEvaluateResults';
import store from '@/store';


const localVue = createLocalVue();


describe('BaseEvaluateResults.vue', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallowMount(BaseEvaluateResults, {
      localVue,
      store,
      mocks: {
        $t: () => 'some specific text',
      },
      propsData: {
        repository: {
          intents_list: [],
          labels_list: [],
          entities_lis: [],
        },
      },
    });
  });

  test('renders correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('Test filter query', () => {
    beforeEach(() => {
      wrapper.vm.onSearch({
        search: 'i like eat sushi',
        intent: 'like',
      });
    });

    test('Valid filter', () => {
      expect(wrapper.vm.query).toMatchObject({
        search: 'i like eat sushi',
        intent: 'like',
      });
    });

    test('Invalid filter', () => {
      expect(wrapper.vm.query).not.toMatchObject({
        search: 'i like eat sushi',
        intent: '',
      });
    });
  });
});
