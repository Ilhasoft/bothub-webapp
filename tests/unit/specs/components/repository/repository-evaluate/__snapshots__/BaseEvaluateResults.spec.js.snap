// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BaseEvaluateResults.vue renders correctly 1`] = `<p>some specific text</p>`;
